pkg/reconciler/ksvc/resources/knative_configuration.go

```diff
@@ -28,6 +29,39 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+func CreateOrchKnativeConfiguration(name string, metadata metav1.ObjectMeta, serviceFQDN string) *knservingv1alpha1.Configuration {
```

Comment:
I think we may need a refactor here. Rather than organizing these by knative component, it may make sense to organize these by kfserving component.

The kfservice_controller.go propagates state across a number of components. Should we do a common interface like this:

pkg/reconciliation/executor/controller.go
``` 
func reconcile(kfsvc KFService) error, KFService.Status {
 // Reconcile Executor resources, return a suggested status to merge 
}
```
pkg/reconciliation/predictor/controller.go
``` 
func reconcile(kfsvc KFService) error, KFService.Status {
 // Reconcile Predictor resources, return a suggested status to merge 
}
```

At the top level, we can just run through an array of controller.reconcile(kfsvc) and merge all the results, and then update the KFService.




for each of the components.

Reply:
Makes sense.

Reply:
I'm not sure if you want to take on this refactor, it may be kind of big. Feel free to punt.